tools/generator.js

```diff
@@ -0,0 +1,163 @@
+    Name: 'Generator',
+
+    _init: function (dir_path) {
+        this._dir_path = dir_path;
+        this._sequence = 0;
+    },
+
+    _create_root: function () {
+        return {
+            'articles': [],
+            'images': [],
+            'sets': []
+        };
+    },
+
+    _create_article: function (path, title, mime_type, category, sequence) {
+        let tags = ['EknArticleObject'];
+
+        if (category)
+            tags.push(category);
+
+        return {
+            'contentType': mime_type,
+            'source': path,
+            'tags': tags,
+            'title': title,
+            'sequenceNumber': sequence,
+        };
+    },
+
+    _create_image: function (path, mime_type) {
+        let license = 'CC-BY 3.0';
+        let width = 0;
+        let height = 0;
+
+        return {
+            'contentType': mime_type,
+            'license': license,
+            'tags': [
+                'EknMediaObject'
+            ],
+            'height': height,
+            'width': width,
+            'source': path
+        };
+    },
+
+    _create_set: function (path, title, child_tags, tags, thumbnail_path) {
+        let _tags = ['EknSetObject'];
+        let featured = false;
+
+        if (tags.length === 0) {
+            _tags.push('EknHomePageTag');
+            featured = true;
+        }
+
+        tags.forEach(tag => {
+            _tags.push(tag);
+        });
+
+        return {
+            'source': path,
+            'childTags': child_tags,
+            'tags': _tags,
+            'featured': featured,
+            'thumbnail': thumbnail_path,
+            'title': title
+        };
+    },
+
+    _convert_recursive: function (dir_path, categories, output) {
+        let directory = Gio.File.new_for_path(dir_path);
+        let enumerator = directory.enumerate_children('*', Gio.FileQueryInfoFlags.NONE, null);
+        let info = enumerator.next_file(null);
+
+        while (info !== null) {
+            let file = enumerator.get_child(info);
+            let file_type = info.get_file_type();
+            let mime_type = info.get_content_type();
+            let path = file.get_path();
+            let name = info.get_name().split('.')[0];
+
+            if (file_type === Gio.FileType.DIRECTORY) {
+                let _categories = categories.slice(0);
+                _categories.push(name);
+
+                this._convert_recursive(path, _categories, output);
+
+                let tags = [];
+                if (categories.length > 0)
+                    tags = [categories[categories.length - 1]];
+
+                let thumbnail_path = path + '/thumbnail.jpg';
+
+                output['sets'].push(this._create_set(path, name, [name], tags, thumbnail_path));
+
+            } else if (mime_type.startsWith('image')) {
+                output['images'].push(this._create_image(path, mime_type));
+
+            } else if (mime_type === 'application/pdf') {
+                this._sequence += 1;
```

Comment:
Yes, I thought about having that name convention as well, but for now I don't think it does any harm having that sequence number as it is (in both cases the order is random and at least it makes it easier to edit on the generated json file in case someone wants to enforce an specific order).